pulp_2to3_migration/app/models/base.py

```diff
@@ -11,24 +17,105 @@ class MigrationPlan(Model):
         plan (models.JSONField): The migration plan in the JSON format
     """
     plan = JSONField()
+    _real_plan = None
+
+    @property
+    def plan_view(self):
+        # Needs a MongoDB connection initialized beforehand to look for importers,
+        # verify existance of repositories and distributors, etc.
+        if not self._real_plan:
+            self._real_plan = ValidatedMigrationPlan(self.plan)
+
+        return self._real_plan
+
+    def get_plugins(self):
+        """
+        Return a list of pulp2 plugins to migrate.
+        """
+        return self.plan_view.plugins_to_migrate
 
     def get_repositories(self):
         """
         Return a list of pulp2 repositories to migrate or empty list if all should be migrated.
         """
-        # TODO: get pulp2 repositories from the self.plan
-        return []
+        return self.plan_view.repositories_to_migrate
 
     def get_importers(self):
         """
         Return a list of pulp2 importers to migrate or empty list if all should be migrated.
         """
-        # TODO: get pulp2 importers from the self.plan
-        return []
+        return self.plan_view.importers_to_migrate
 
     def get_distributors(self):
         """
         Return a list of pulp2 distributors to migrate or empty list if all should be migrated.
         """
-        # TODO: get pulp2 distributors from the self.plan
-        return []
+        return self.plan_view.distributors_to_migrate
+
+
+class SchemaValidationFailure(Exception):
+    pass
+
+
+class ValidatedMigrationPlan:
+    def __init__(self, migration_plan):
+        self.migration_plan = migration_plan
+
+        self.plugins_to_migrate = []
+        self.importers_to_migrate = []
+        self.distributors_to_migrate = []
+        self.repositories_to_migrate = []
+
+        self.validate_and_populate()
+
+    def validate_and_populate(self):
+        for plugin_data in self.migration_plan['plugins']:
+            self.plugins_to_migrate.append(plugin_data['type'])
+            if plugin_data.get('repositories'):
+                self._parse_repository_data(plugin_data.get('repositories'))
+            # protection
+
+    def _parse_repository_data(self, repository_data):
+        for repository in repository_data:
```

Comment:
I'm not doing anything with  "name" here - I'm still not sure what it is for exactly.  Is it the name of the Pulp 3 repository to be created?  Or does it relate to Pulp 2?